src/elasticsearch/index.ts

```diff
@@ -1,166 +1,440 @@
 import getFunctionScores from './score'
 import getMultiMatchConfig from './multimatch'
 import getBoosts from './boost'
-import getMapping from './mapping'
 import cloneDeep from 'clone-deep'
 import SearchQuery from '../types/SearchQuery'
+import AppliedFilter from '../types/AppliedFilter'
 import ElasticsearchQueryConfig from './types/ElasticsearchQueryConfig'
 
-function processNestedFieldFilter (attribute: string, value: any) {
-  let processedFilter = {
-    'attribute': attribute,
-    'value': value
-  };
-  let filterAttributeKeys = Object.keys(value);
-  for (let filterAttributeKey of filterAttributeKeys) {
-    if (value[filterAttributeKey] && !Array.isArray(value[filterAttributeKey]) && typeof value[filterAttributeKey] === 'object') {
-      processedFilter = processNestedFieldFilter(attribute + '.' + filterAttributeKey, value[filterAttributeKey]);
-    }
-  }
-  return processedFilter;
+export interface FilterOptions {
+  operator: string
+  attribute: string
+  value: any,
+  queryChain?: any
+}
+
+export interface FilterInterface {
+  check ({ operator, value, attribute, queryChain }: FilterOptions): boolean,
+  filter ({ operator, value, attribute, queryChain }: FilterOptions): Object
+  mutator? (value: any): any,
+  priority?: number
+}
+
+export interface FiltersInterface {
+  [key: string]: FilterInterface
 }
 
 /**
- *
- * @param {Object} object
- * @param {String} scope
- * @returns {boolean}
+ * Class to create filter in modular and reproductive way
  */
-function checkIfObjectHasScope ({ object, scope }: { object: { scope: string }, scope: string}) {
-  return object.scope === scope || (Array.isArray(object.scope) && object.scope.indexOf(scope) >= 0);
-}
+export default class Filters {
```

Comment:
what do you think about moving this into separete file?

Reply:
Done